### PR TITLE
Postprocessing to modify plans returned by llm planner; minor issues with query-ui

### PR DESCRIPTION
**Pull request by amolvdeshpande**
1. Added a post-processing step where the plan returned by the LLM Planner is modified if it uses a "QueryVectorDatabase" in the beginning, followed by a "count" or "extract_entity". Instead, we use a "QueryDatabase" followed by an LlmFilter. 

2. Fixed a couple of minor issues in query-ui so that "Execute Code" works and prints results for queries with a single answer.

### CODE REVIEW

File: lib/sycamore/sycamore/query/planner.py
Comment (baitsguy) [inline lib/sycamore/sycamore/query/planner.py:150]: we'll want to use the llm object from the class here so we can support non-openai cases

File: lib/sycamore/sycamore/query/planner.py
Comment (baitsguy) [inline lib/sycamore/sycamore/query/planner.py:187]: might be better to create the new_op dict first and only set "query" based on the condition, saves us from mistakes if adding/removing fields

File: lib/sycamore/sycamore/query/planner.py
Comment (baitsguy) [inline lib/sycamore/sycamore/query/planner.py:203]: does the break in the word "records" matter here

File: lib/sycamore/sycamore/query/planner.py
Comment (baitsguy) [inline lib/sycamore/sycamore/query/planner.py:205]: should we say this is an example instead of saying its the format

File: lib/sycamore/sycamore/tests/unit/query/test_plan.py
Comment (baitsguy) [inline lib/sycamore/sycamore/tests/unit/query/test_plan.py:161]: can we add a test for post_process as well

File: lib/sycamore/sycamore/query/planner.py
Comment (baitsguy) [inline lib/sycamore/sycamore/query/planner.py:81]: nit: we're using `post_process` style naming, also can you default to True? I think we always want this now

**Review by baitsguy — COMMENTED** [review]
Looks good, a few small questions/nits and requested a unit test

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py]: Fixed to use llm_client from the class.

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py]: Done.

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py]: Fixed.

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py]: Fixed.

File: lib/sycamore/sycamore/tests/unit/query/test_plan.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/tests/unit/query/test_plan.py]: Added unit tests.

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py]: Remove postprocess as a flag to process_json_plan (which was added to handle unit tests for that function).

**Review by baitsguy — APPROVED** [review]
We can work on writing a separate optimizer as we build a techniques but this should get us started

File: apps/query-ui/queryui/Sycamore_Query.py
Comment (mdwelsh) [inline apps/query-ui/queryui/Sycamore_Query.py:72]: This is only in the case where we did code generation, right?

File: lib/sycamore/sycamore/query/planner.py
Comment (mdwelsh) [inline lib/sycamore/sycamore/query/planner.py:169]: I'm very surprised you need an LLM call for this. Can you explain?

File: lib/sycamore/sycamore/query/planner.py
Comment (mdwelsh) [inline lib/sycamore/sycamore/query/planner.py:189]: Again, surprised we need an LLM call to do this. Can't we just add the LLM filter op directly?

File: lib/sycamore/sycamore/query/planner.py
Comment (mdwelsh) [inline lib/sycamore/sycamore/query/planner.py:224]: Since node IDs don't matter and don't need to be in sequence, why not just generate a new node ID that is the max of the existing node IDs plus 1, and use that. That way we don't need to tweak the others.

File: lib/sycamore/sycamore/tests/unit/query/test_postprocess_plan.py
Comment (mdwelsh) [inline lib/sycamore/sycamore/tests/unit/query/test_postprocess_plan.py:66]: Remove this line.

File: lib/sycamore/sycamore/query/planner.py
Comment (baitsguy) [reply lib/sycamore/sycamore/query/planner.py:169]: it's just changing the description which can say this is a vector search, not the actual node object

File: apps/query-ui/queryui/Sycamore_Query.py
Comment (amolvdeshpande) [reply apps/query-ui/queryui/Sycamore_Query.py:72]: Correct. If I understand the flow here correctly, query_id wouldn't be generated in this case.

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py:169]: As Vinayak said, we need to convert the original description like "find all incidents that were caused by environmental conditions" to "find all incidents" -- assuming we want the description to be correct, an LLM would have to be used to manipulate the text properly..

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py:189]: Same reason as above. The query phrase is something like "wildfires" -- needs to be converted to a boolean question like "Does this incident involve wildfires" for the LLM to answer.

File: lib/sycamore/sycamore/query/planner.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py:224]: I feel like them being in sequence might be useful later (e.g., when we are drawing the plans or something like that) and it's not tricky to do it. But I can change it.

File: lib/sycamore/sycamore/tests/unit/query/test_postprocess_plan.py
Comment (amolvdeshpande) [reply lib/sycamore/sycamore/tests/unit/query/test_postprocess_plan.py]: Done.

**Review by baitsguy — APPROVED** [review]

File: lib/sycamore/sycamore/query/planner.py
Comment (baitsguy) [inline lib/sycamore/sycamore/query/planner.py:126]: curious why you're using model_validate and a dict as opposed to just create a regular object directly?

Comment (amolvdeshpande) [reply lib/sycamore/sycamore/query/planner.py:126]: Primarily to stay consistent with how they are being created in the other code right now and not duplicate functionality. Also, I didn't see an appropriate __init__ function, but perhaps it is being created implicitly? I can change it to direct object creation if that's the more appropriate way to do it.

Comment (baitsguy) [reply lib/sycamore/sycamore/query/planner.py:126]: You should just be able to initialize them like so: https://github.com/aryn-ai/sycamore/blob/0e2a9f5e586474c15bbfd1f85cf8b3bc3a1e4543/lib/sycamore/sycamore/tests/unit/query/test_planner.py#L68

I had suggested since we're creating the object ourselves as opposed to trying to create one from another data source (like the llm giving us a json).

Don't block on it though, it's your call if trivial to do

**Pull request merged**
